pkg/xds/bootstrap/generator.go

```diff
@@ -28,29 +34,46 @@ type BootstrapGenerator interface {
 func NewDefaultBootstrapGenerator(
 	resManager core_manager.ResourceManager,
 	config *bootstrap_config.BootstrapParamsConfig,
-	cacertFile string) BootstrapGenerator {
+	cacertFile string,
+	environmentType core.EnvironmentType) BootstrapGenerator {
 	return &bootstrapGenerator{
-		resManager:  resManager,
-		config:      config,
-		xdsCertFile: cacertFile,
+		resManager:      resManager,
+		config:          config,
+		xdsCertFile:     cacertFile,
+		environmentType: environmentType,
 	}
 }
 
 type bootstrapGenerator struct {
-	resManager  core_manager.ResourceManager
-	config      *bootstrap_config.BootstrapParamsConfig
-	xdsCertFile string
+	resManager      core_manager.ResourceManager
+	config          *bootstrap_config.BootstrapParamsConfig
+	xdsCertFile     string
+	environmentType core.EnvironmentType
 }
 
 func (b *bootstrapGenerator) Generate(ctx context.Context, request types.BootstrapRequest) (proto.Message, error) {
 	proxyId, err := core_xds.BuildProxyId(request.Mesh, request.Name)
 	if err != nil {
 		return nil, err
 	}
-	dataplane, err := b.fetchDataplane(ctx, proxyId)
-	if err != nil {
-		return nil, err
+	var dataplane *core_mesh.DataplaneResource
+	switch b.environmentType {
+	case core.KubernetesEnvironment:
+		if dataplane, err = b.fetchDataplane(ctx, proxyId); err != nil {
+			return nil, err
+		}
+	case core.UniversalEnvironment:
+		dpYAML, err := base64.StdEncoding.DecodeString(request.DataplaneResource)
+		if err != nil {
+			return nil, err
+		}
+		if dataplane, err = core_mesh.ParseDataplaneYAML(dpYAML); err != nil {
+			return nil, err
+		}
+	default:
+		return nil, errors.Errorf("unknown environment type %s", b.environmentType)
 	}
+
```

Comment:
I think it might be beneficial to support old flow on universal, therefore this func should be: if dataplane resource was passed in the request, use it, if not, try to fetch it from storage. This way you don't have to switch over environment.